plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -115,4 +206,113 @@ protected void deleteTableDirectory(
+            }
+            return Optional.of(definition);
+        }
+        return Optional.empty();
+
+    }
+
+    protected Map<String, String> createViewProperties(ConnectorSession session)
+    {
+        return ImmutableMap.<String, String>builder()
+                .put(PRESTO_VIEW_FLAG, "true")
+                .put(TRINO_CREATED_BY, TRINO_CREATED_BY_VALUE)
+                .put(PRESTO_VERSION_NAME, trinoVersion)
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
+                .put(TABLE_COMMENT, PRESTO_VIEW_COMMENT)
+                .buildOrThrow();
+    }
+
+    protected SchemaTableName createMaterializedViewStorageTable(ConnectorSession session, SchemaTableName schemaViewName, ConnectorMaterializedViewDefinition definition)
+    {
+        // Generate a storage table name and create a storage table. The properties in the definition are table properties for the
+        // storage table as indicated in the materialized view definition.
+        String storageTableName = "st_" + randomUUID().toString().replace("-", "");
+        Map<String, Object> storageTableProperties = new HashMap<>(definition.getProperties());
+        storageTableProperties.putIfAbsent(FILE_FORMAT_PROPERTY, DEFAULT_FILE_FORMAT_DEFAULT);
+
+        SchemaTableName storageTable = new SchemaTableName(schemaViewName.getSchemaName(), storageTableName);
+        List<ColumnMetadata> columns = definition.getColumns().stream()
+                .map(column -> new ColumnMetadata(column.getName(), typeManager.getType(column.getType())))
+                .collect(toImmutableList());
+
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(storageTable, columns, storageTableProperties, Optional.empty());
+        Transaction transaction = IcebergUtil.newCreateTableTransaction(this, tableMetadata, session);
+        transaction.newAppend().commit();
+        transaction.commitTransaction();
+        return storageTable;
+    }
+
+    protected ConnectorMaterializedViewDefinition getMaterializedViewDefinition(
+            SchemaTableName schemaViewName,
+            Table icebergTable,
+            Optional<String> owner,
+            String viewOriginalText,
+            String storageTableName)
+    {
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        properties.put(FILE_FORMAT_PROPERTY, IcebergUtil.getFileFormat(icebergTable));
+        if (!icebergTable.spec().fields().isEmpty()) {
+            properties.put(PARTITIONING_PROPERTY, toPartitionFields(icebergTable.spec()));
+        }
+
+        IcebergMaterializedViewDefinition definition = decodeMaterializedViewData(viewOriginalText);
+        return new ConnectorMaterializedViewDefinition(
+                definition.getOriginalSql(),
+                Optional.of(new CatalogSchemaTableName(catalogName.toString(), new SchemaTableName(schemaViewName.getSchemaName(), storageTableName))),
+                definition.getCatalog(),
```

Comment:
Isn't it safer to always take the catalogName here? If the catalog is renamed it would still work as expected.